src/lib/jsonParseV2/parseBatchUpdate.cpp

```diff
@@ -113,7 +113,19 @@ std::string parseBatchUpdate(ConnectionInfo* ciP, BatchUpdate* burP)
     }
     else if (name == "actionType")
     {
-      burP->updateActionType.set(iter->value.GetString());
+      UpdateActionType uat;
+      uat.set(iter->value.GetString());
+
```

Comment:
Why create an additional vartiable of type `UpdateActionType` ?
`burP->updateActionType` could be used instead.
Less code and slightly faster.

Also, the set() method of UpdateActionType could call its check() method and return false if check() fails:

    if (burP->updateActionType.set(iter->value.GetString(), &err) == false)
    {
      alarmMgr.badInput(clientIp, err);
      ...
    }